src/main/java/org/openstreetmap/atlas/checks/validation/tag/HighwayMissingNameAndRefTagCheck.java

```diff
@@ -0,0 +1,241 @@
+package org.openstreetmap.atlas.checks.validation.tag;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.openstreetmap.atlas.checks.base.BaseCheck;
+import org.openstreetmap.atlas.checks.flag.CheckFlag;
+import org.openstreetmap.atlas.geography.Heading;
+import org.openstreetmap.atlas.geography.atlas.items.AtlasObject;
+import org.openstreetmap.atlas.geography.atlas.items.Edge;
+import org.openstreetmap.atlas.geography.atlas.items.Node;
+import org.openstreetmap.atlas.geography.atlas.walker.OsmWayWalker;
+import org.openstreetmap.atlas.tags.HighwayTag;
+import org.openstreetmap.atlas.tags.JunctionTag;
+import org.openstreetmap.atlas.tags.names.NameTag;
+import org.openstreetmap.atlas.tags.names.ReferenceTag;
+import org.openstreetmap.atlas.utilities.configuration.Configuration;
+import org.openstreetmap.atlas.utilities.scalars.Angle;
+
+/**
+ * Auto generated Check template
+ *
+ * @author v-garei
+ */
+public class HighwayMissingNameAndRefTagCheck extends BaseCheck<Long>
+{
+    private static final long serialVersionUID = 8198312161814763037L;
+    private static final String MISSING_BOTH_NAME_AND_REF_TAG_INSTRUCTIONS = "Way {0, number, #} is missing both name and ref tag. Way must contain either one.";
+    private static final List<String> FALLBACK_INSTRUCTIONS = Collections
+            .singletonList(MISSING_BOTH_NAME_AND_REF_TAG_INSTRUCTIONS);
+    private final double minAngleForNonContiguousWays;
+    private static final double MIN_ANGLE_NON_CONTIGUOUS_WAYS = 30;
+
+    /**
+     * The default constructor that must be supplied. The Atlas Checks framework will generate the
+     * checks with this constructor, supplying a configuration that can be used to adjust any
+     * parameters that the check uses during operation.
+     *
+     * @param configuration
+     *            the JSON configuration for this check
+     */
+    public HighwayMissingNameAndRefTagCheck(final Configuration configuration)
+    {
+        super(configuration);
+        this.minAngleForNonContiguousWays = configurationValue(configuration,
+                "min.contiguous.angle", MIN_ANGLE_NON_CONTIGUOUS_WAYS);
+    }
+
+    /**
+     * This function will validate if the supplied atlas object is valid for the check.
+     *
+     * @param object
+     *            the atlas object supplied by the Atlas-Checks framework for evaluation
+     * @return {@code true} if this object should be checked
+     */
+    @Override
+    public boolean validCheckForObject(final AtlasObject object)
+    {
+
+        // by default we will assume all objects as valid
+        return object instanceof Edge && ((Edge) object).isMainEdge()
+                && !isFlagged(object.getOsmIdentifier())
+                && ((Edge) object).highwayTag().isMoreImportantThanOrEqualTo(HighwayTag.TERTIARY)
```

Comment:
It would be nice to have this highway tag value be configurable.

Reply:
addressed